### PR TITLE
Reprovision group members (CO Persons) when fixing Unix Cluster groups (SOFTWARE-52633)

### PR DESCRIPTION
**Pull request by brianaydemir**
Intended to address Issue #14.

### CODE REVIEW

**Comment by brianaydemir** [comment]
Tested this on the "test" COmanage/CILogon instance. Note the `gidNumber` attribute. (`query_ldap_dev.py` is a personal utility script. Some output is elided in the name of privacy.)

### Before

```
$ ./query_ldap_dev.py '(uid=http://cilogon.org/serverB/users/47453732)'
[DN: voPersonID=OSG1000029,ou=people,o=OSG,o=CO,dc=cilogon,dc=org - STATUS: Read - READ TIME: 2023-07-25T14:14:13.887804
    gidNumber: 10034
    isMemberOf: CO:members:all
                CO:members:active
                httpcilogon.orgserverbusers47453732.unknownname UnixCluster Group
    uid: http://cilogon.org/serverB/users/47453732
    uidNumber: 10036
]
```

Provisioned: Thu Aug 18 09:46:44 2022 America/Los_Angeles

### After

```
$ PASS="..." ./group_fixup.py -u co_8.baydemir_test -c 8 -e https://registry-test.cilogon.org/registry/ -p 9 -x 2695
CO 8 Group 2695: "httpcilogon.orgserverbusers47453732.unknownname"
   - Identifier 2959: (osggroup) "httpcilogon.orgserverbusers47453732.unknownname"
   - Identifier 2960: (osggid) "10036"

:thumbsup:

$ ./query_ldap_dev.py '(uid=http://cilogon.org/serverB/users/47453732)'
[DN: voPersonID=OSG1000029,ou=people,o=OSG,o=CO,dc=cilogon,dc=org - STATUS: Read - READ TIME: 2023-07-25T14:15:29.478217
    gidNumber: 10036
    isMemberOf: CO:members:all
                CO:members:active
                httpcilogon.orgserverbusers47453732.unknownname
    uid: http://cilogon.org/serverB/users/47453732
    uidNumber: 10036
]
```

Provisioned: Tue Jul 25 14:14:58 2023 America/Los_Angeles

File: group_fixup.py [hunk 1]
```diff
@@ -276,6 +276,22 @@ def provision_group(gid):
     return call_api3(POST, path, data)
 
 
+def provision_group_members(gid):
+    prov_id = options.prov_id
+    def path(pid): return f"co_provisioning_targets/provision/{prov_id}/copersonid:{pid}.json"
```
Comment (brianhlin) [inline group_fixup.py]:
```suggestion
    def path(pid, prov_id): return f"co_provisioning_targets/provision/{prov_id}/copersonid:{pid}.json"
```

For sanity's sake, `prov_id` should probably be a positional arg as well. Though if this is just a function that's only used in one spot that just returns an f-string, it feels like we should just use the f-string at the call site.

File: group_fixup.py [hunk 2]
```diff
@@ -276,6 +276,22 @@ def provision_group(gid):
     return call_api3(POST, path, data)
 
 
+def provision_group_members(gid):
+    prov_id = options.prov_id
```
Comment (brianhlin) [inline group_fixup.py:280]:
```suggestion
def provision_group_members(gid, prov_id):
```

It doesn't look like `options` is global so I think this should be a function arg

**Review by brianhlin — CHANGES_REQUESTED** [review]
Some minor changes requested but otherwise LGTM

File: group_fixup.py (hunk 2 above)
Comment (brianaydemir) [reply group_fixup.py:280]: That bit is a direct copy of `provision_group` (`options` is defined on line 74).

I kinda favor stylistic consistency over the global reference, even though I know of at least one time that this global reference style caused the originally author himself some confusion during testing.

File: group_fixup.py (hunk 1 above)
Comment (brianaydemir) [reply group_fixup.py]: I agree, it's less strange to define `path` as an f-string right before the API call. I've pushed that change.

(See my other response re: `options` and `prov_id`.)

File: group_fixup.py (hunk 2 above)
Comment (brianhlin) [reply group_fixup.py:280]: 😿 I suppose I can live with that

**Review by brianhlin — APPROVED** [review]
LGTM

**Pull request merged**